### PR TITLE
modified task creation form

### PR DESCRIPTION
**Pull request by deusalexmachina**
1. Validate using WTForms for task creation 
2. split datetime-local into date and time fields for browser compatibility 
3. deleted unused relationships in user and task models
4. updated travis.yml, added codecov, heroku deploy
5. added .flake8 to ignore migrations folder during CI
6. brought test coverage to 83%

### CODE REVIEW

File: task_manager/routes/tasks.py [hunk 1]
```diff
@@ -1,9 +1,11 @@
-from flask import request, make_response, render_template
+from flask import request, make_response, render_template, url_for, flash
```
Comment (SemicolonExpected) [inline task_manager/routes/tasks.py:1]: F401: 'flask.flash' imported but unused

**Review by SemicolonExpected — CHANGES_REQUESTED** [review]
[Lintly](https://github.com/grantmcconnaughey/Lintly) has detected code quality issues in this pull request.

File: task_manager/routes/tasks.py [hunk 2]
```diff
@@ -1,9 +1,11 @@
-from flask import request, make_response, render_template
+from flask import request, make_response, render_template, url_for
```
Comment (SemicolonExpected) [inline task_manager/routes/tasks.py:1]: F401: 'flask.request' imported but unused

**Review by SemicolonExpected — CHANGES_REQUESTED** [review]
[Lintly](https://github.com/grantmcconnaughey/Lintly) has detected code quality issues in this pull request.

File: task_manager/routes/tasks.py (hunk 2 above)
Comment (SemicolonExpected) [inline task_manager/routes/tasks.py:1]: F401: 'flask.request' imported but unused

**Review by SemicolonExpected — CHANGES_REQUESTED** [review]
[Lintly](https://github.com/grantmcconnaughey/Lintly) has detected code quality issues in this pull request.

File: task_manager/tests/test_user.py [hunk 3]
```diff
@@ -1,6 +1,6 @@
-
 import unittest
-# from flask_login import current_user, login_user, logout_user
+
+from flask import url_for
```
Comment (SemicolonExpected) [inline task_manager/tests/test_user.py:3]: F401: 'flask.url_for' imported but unused

File: task_manager/tests/test_auth.py [hunk 4]
```diff
@@ -0,0 +1,62 @@
...
+
+    def test_invalid_login(self):
+        '''TEST INVALID LOGIN'''
+        user = User(username='testlogin', email='testlogin@test.com',
+                    password='pass')
+        user.set_password('pass123')
+        user.save_user()
+        response = self.login('testlogin', 'pass')
+        self.assertEqual(response.status_code, 200)
+
```
Comment (SemicolonExpected) [inline task_manager/tests/test_auth.py:62]: W391: blank line at end of file

File: task_manager/routes/users.py [hunk 5]
```diff
@@ -30,3 +25,7 @@ def delete_user(user_id):
         user = User.query.filter_by(id=int(user_id)).first_or_404()
         user.delete_user()
         return redirect(url_for('index'))
+
+
+def update_user(user_id):
+    return None
```
Comment (SemicolonExpected) [inline task_manager/routes/users.py:31]: W292: no newline at end of file

File: task_manager/__init__.py [hunk 6]
```diff
@@ -2,7 +2,7 @@
 from flask_login import LoginManager
 from flask_marshmallow import Marshmallow
 from flask_migrate import Migrate
-from flask_restx import Api
+from flask_restx import Api, Namespace
```
Comment (SemicolonExpected) [inline task_manager/__init__.py:5]: F401: 'flask_restx.Namespace' imported but unused

**Review by SemicolonExpected — CHANGES_REQUESTED** [review]
[Lintly](https://github.com/grantmcconnaughey/Lintly) has detected code quality issues in this pull request.

File: task_manager/routes/tasks.py [hunk 7]
```diff
@@ -79,50 +61,38 @@ def model_get_update_task(task_id):
...
+            task.priority = form.priority.data
+            task.start_time = form.start
+            task.end_time = form.end
+
+        except Exception as e:
+            print(e)
+            db.session.rollback()
+        else:
+            db.session.commit()
+            return redirect(f'/dashboard')
```
Comment (SemicolonExpected) [inline task_manager/routes/tasks.py:90]: F541: f-string is missing placeholders

File: task_manager/routes/tasks.py [hunk 8]
```diff
@@ -16,55 +18,35 @@ class Meta:
...
+            new_assignment = Assignment(time_added=datetime.today(),
+                                        user_id=current_user.id,
+                                        task_id=new_task.id)
+            db.session.add(new_assignment)
+        except Exception as e:
+            print(e)
+            db.session.rollback()
+        else:
+            db.session.commit()
+            return redirect(f'/dashboard')
```
Comment (SemicolonExpected) [inline task_manager/routes/tasks.py:47]: F541: f-string is missing placeholders

**Review by SemicolonExpected — CHANGES_REQUESTED** [review]
[Lintly](https://github.com/grantmcconnaughey/Lintly) has detected code quality issues in this pull request.

File: task_manager/api.py [hunk 9]
```diff
@@ -7,130 +7,130 @@
...
-		return "Hello World"
+    def get(self):
+        return "Hello World"
 
 
 @apis.route('/')
 @apis.route('/dashboard')  # dashboard
 class Index(Resource):
-	'''If logged in display index, else display homepage'''
+    '''If logged in display index, else display homepage'''
```
Comment (SemicolonExpected) [inline task_manager/api.py:23]: is this a spaces vs tabs thing?

File: task_manager/forms.py [hunk 10]
```diff
@@ -1,23 +1,63 @@
...
+    def start(self):
+        if self.start_date.data and self.start_time.data:
+            return datetime.combine(self.start_date.data, self.start_time.data)
+
+    @property
+    def end(self):
+        if self.end_date.data and self.end_time.data:
+            return datetime.combine(self.end_date.data, self.end_time.data)
+
+    # def validate_end_date(self, field):
```
Comment (SemicolonExpected) [inline task_manager/forms.py:34]: how does this validate the endtime > starttime?

File: task_manager/tests/test_task.py [hunk 11]
```diff
@@ -3,61 +3,76 @@
...
-		'''
-		self.assertEqual('1', '1')
-
-	def test_Delete(self):
-		'''
-		TEST DELETE TASK
-		'''
-		self.assertEqual('1', '1')
+    def setUp(self):
+        self.app = create_app('config.DevelopmentConfig')
```
Comment (SemicolonExpected) [inline task_manager/tests/test_task.py:11]: can you summarize the changes in this file here?

File: task_manager/forms.py (hunk 10 above)
Comment (deusalexmachina) [reply task_manager/forms.py:34]: i commented this out since it was interfering with a testcase. i didn't have enough time to debug and had to get our coverage up from 50%.

File: task_manager/tests/test_task.py (hunk 11 above)
Comment (deusalexmachina) [reply task_manager/tests/test_task.py:11]: added test client and config parameters. tests basically pass context to a view and checks for assertion equivalence with the expected http status code

File: task_manager/api.py (hunk 9 above)
Comment (deusalexmachina) [reply task_manager/api.py:23]: flake8 was throwing lint errors since our file had mix of spaces and tabs. i ran a global cleanup to convert all tabs to spaces according to flake8 convention.

Comment (SemicolonExpected) [reply task_manager/api.py:23]: in the future I would prefer all the spaces be turned into tabs, our flake8 ignores space errors because tabs were preferred
